source/configuration/sharding.txt

```diff
@@ -0,0 +1,161 @@
+.. _mongoid-sharding-configuration:
+
+======================
+Sharding Configuration
+======================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: ruby framework, odm, code example
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Sharding is a way to distribute your data across multiple machines. MongoDB
+uses sharding to support deployments with large data sets and high
+throughput operations. In this guide, you can learn how to configure sharding in
+your {+odm+} application.
+
+Declare Shard keys
+------------------
+
+MongoDB uses shard keys to distribute a collection's documents across
+shards. A shard key consists of one or more fields in a document. In your
+{+odm+} application, you can declare a shard key when you create a model by
+using the ``shard_key`` macro.
+
+The following example creates a ``Person`` class with a shard key on the
+``ssn`` field:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key
+   :end-before: # end-shard-key
+   :emphasize-lines: 6
+
+.. note::
+
+   To shard a collection, the collection must have an index that starts with the
+   shard key. You can use {+odm+}'s index management functionality to create the
+   index. To learn more about index management with {+odm+}, see the :ref:`Index
+   Management <mongoid-indexes>` guide.
+
+If a model declares a shard key, {+odm+} expects the respective collection to be
+sharded with the specified shard key. When {+odm+} reloads models, it provides
+the shard key with the ``_id`` field to the ``find`` command to improve query
+performance. If the collection is not sharded with the specified shard key,
+queries might not return the expected results.
+
+You can declare shard keys by using either the standard MongoDB
```

Comment:
q: does the manual call these 'standard' and 'shorthand' syntaxes? if not, maybe 'full syntax' or something similar would be more accurate. applies throughout.